Mathlib/MeasureTheory/Function/L2Space.lean

```diff
@@ -256,6 +256,36 @@ theorem inner_indicatorConstLp_one (hs : MeasurableSet s) (hμs : μ s ≠ ∞)
     ⟪indicatorConstLp 2 hs hμs (1 : 𝕜), f⟫ = ∫ x in s, f x ∂μ := by
   rw [L2.inner_indicatorConstLp_eq_inner_setIntegral 𝕜 hs hμs (1 : 𝕜) f]; simp
 
+local notation "⟪" x ", " y "⟫" => @inner ℝ _ _ x y
+
+/- The inner product in `L2` of multiples `a` and `b` of indicators of two sets with finite measure
+is `a * b` times the measure of the intersection. -/
+lemma inner_indicatorConstLp_indicatorConstLp
+ {v w : (Set α)} (hv : MeasurableSet v)
+  (hw : MeasurableSet w) (hμv : μ v ≠ ∞) (hμw : μ w ≠ ∞) (a b : ℝ):
+  ⟪((indicatorConstLp 2 hv hμv (a : ℝ))), (indicatorConstLp 2 hw hμw (b : ℝ))⟫ =
+    a * b * (μ.real (v ∩ w)) := by
```

Comment:
```suggestion
lemma inner_indicatorConstLp_indicatorConstLp {v w : Set α} (hv : MeasurableSet v)
    (hw : MeasurableSet w) (hμv : μ v ≠ ∞ := by finiteness) (hμw : μ w ≠ ∞ := by finiteness) 
    (a b : ℝ) :
    ⟪indicatorConstLp 2 hv hμv a, indicatorConstLp 2 hw hμw b⟫ = a * b * μ.real (v ∩ w) := by
```
I've fixed the indentation, spurious spaces and useless parentheses.

I've also added back `:= by finiteness`. What that means is that, when applying the lemma, the user won't need in some cases to provide the proof of `hμv` and `hμw`, this will done under the hood by the system (for instance if mu is a finite measure). This is very useful!